frontend/src/settings/beta/SettingsProfileTab.tsx

```diff
@@ -0,0 +1,95 @@
+import ContentCopy from '@mui/icons-material/ContentCopy'
+import Box from '@mui/material/Box'
+import Button from '@mui/material/Button'
+import Chip from '@mui/material/Chip'
+import Divider from '@mui/material/Divider'
+import IconButton from '@mui/material/IconButton'
+import Stack from '@mui/material/Stack'
+import { useTheme } from '@mui/material/styles'
+import Tooltip from '@mui/material/Tooltip'
+import Typography from '@mui/material/Typography'
+import { useState } from 'react'
+import { User } from 'types/types'
+
+function SettingsProfileTab({ user }: { user: User }) {
+  const theme = useTheme()
+  const [displayToken, setDisplayToken] = useState(false)
+  const [displayedToken, setDisplayedToken] = useState('')
+
+  const regenerateToken = async () => {
+    const { token } = await fetch('/api/v1/user/token', {
+      method: 'POST',
+    }).then((res) => res.json())
+
+    return token
+  }
+
+  const showToken = async () => {
+    const token = await regenerateToken()
+    setDisplayedToken(token)
+    setDisplayToken(true)
+  }
+
+  return (
+    <>
+      <Box sx={{ pt: 0, pr: 2, pb: 2, pl: 2 }}>
+        <Typography variant='body1' sx={{ pr: 1, fontWeight: 'medium' }}>
+          Name
+        </Typography>
+        <Divider sx={{ pt: 1, mb: 1 }} />
+        <Typography variant='body1'>{user.id}</Typography>
+      </Box>
+      <Box sx={{ p: 2 }}>
+        <Typography variant='body1' sx={{ pr: 1, fontWeight: 'medium' }}>
+          Roles
+        </Typography>
+        <Divider sx={{ pt: 1, mb: 1 }} />
```

Comment:
Bit picky, but can we pick either padding or margin here. It doesn't matter which one in this case, but is a bit odd to use one for the top and another for the bottom.